### PR TITLE
For Var processing changes in ci.common testing 

### PR DESCRIPTION
**Pull request by arunvenmany-ibm**

### CODE REVIEW

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy [hunk 1]
```diff
@@ -505,10 +505,9 @@ abstract class AbstractServerTask extends AbstractLibertyTask {
         configureMultipleAppsConfigDropins(serverNode)
     }
 
-    protected ServerConfigDocument getServerConfigDocument(CommonLogger log, File serverXML, File configDir, File bootstrapFile,
-            Map<String, String> bootstrapProp, File serverEnvFile, boolean giveConfigDirPrecedence, Map<String, File> libertyDirPropertyFiles) throws IOException {
-        if (scd == null || !scd.getServerXML().getCanonicalPath().equals(serverXML.getCanonicalPath())) {
-            scd = new ServerConfigDocument(log, serverXML, configDir, bootstrapFile, bootstrapProp, serverEnvFile, giveConfigDirPrecedence, libertyDirPropertyFiles)
+    protected ServerConfigDocument getServerConfigDocument(CommonLogger log, Map<String, File> libertyDirPropertyFiles) throws IOException {
+        if (scd == null) {
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy:509]: So this seems problematic. Before we were checking to see if the path for the server.xml file changed and if so we recreated the ServerConfigDocument. Now we cannot tell if it changed. Makes me wonder if we should always create a new ServerConfigDocument, but that is a lot of processing that happens.

File: src/main/groovy/io/openliberty/tools/gradle/tasks/StartTask.groovy [hunk 2]
```diff
@@ -86,8 +86,7 @@ class StartTask extends AbstractServerTask {
         if (serverConfigFile != null && serverConfigFile.exists()) {
             try {
                 Map<String,String> props = combinedBootstrapProperties == null ? convertPropertiesToMap(server.bootstrapProperties) : combinedBootstrapProperties;
-                getServerConfigDocument(new CommonLogger(project), serverConfigFile, server.configDirectory, server.bootstrapPropertiesFile, props, server.serverEnvFile, 
-                                                                            false, getLibertyDirectoryPropertyFiles(null));
+                scd = getServerConfigDocument(new CommonLogger(project), getLibertyDirectoryPropertyFiles(null));
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/StartTask.groovy:89]: I don't think the line before this needs to be done anymore. The bootstrap.properties file in the server dir will be used and should have all the combined properties already.

File: src/main/groovy/io/openliberty/tools/gradle/tasks/UndeployTask.groovy [hunk 3]
```diff
@@ -79,8 +79,7 @@ class UndeployTask extends AbstractServerTask {
             File serverXML = new File(getServerDir(project).getCanonicalPath(), "server.xml")
 
             try {
-                getServerConfigDocument(new CommonLogger(project), serverXML, server.configDirectory,
-                        server.bootstrapPropertiesFile, combinedBootstrapProperties, server.serverEnvFile, false, getLibertyDirectoryPropertyFiles(null))
+                scd= getServerConfigDocument(new CommonLogger(project), getLibertyDirectoryPropertyFiles(null))
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/UndeployTask.groovy:82]: I don't think you need to get serverXML at line 79 any more.

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy [hunk 4]
```diff
@@ -682,8 +682,7 @@ class DeployTask extends AbstractServerTask {
                 File serverXML = new File(getServerDir(project).getCanonicalPath(), "server.xml")
 
                 try {
-                    scd = getServerConfigDocument(new CommonLogger(project), serverXML, server.configDirectory,
-                            server.bootstrapPropertiesFile, combinedBootstrapProperties, server.serverEnvFile, false, getLibertyDirectoryPropertyFiles(null))
+                    scd = getServerConfigDocument(new CommonLogger(project), getLibertyDirectoryPropertyFiles(null))
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy:685]: You don't need to get serverXML anymore at line 682.

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy [hunk 5]
```diff
@@ -520,8 +519,7 @@ abstract class AbstractServerTask extends AbstractLibertyTask {
         if (serverConfigFile != null && serverConfigFile.exists()) {
             try {
                 Map<String,String> props = combinedBootstrapProperties == null ? convertPropertiesToMap(server.bootstrapProperties) : combinedBootstrapProperties;
-                getServerConfigDocument(new CommonLogger(project), serverConfigFile, server.configDirectory, server.bootstrapPropertiesFile, props, server.serverEnvFile, 
-                                                                            false, getLibertyDirectoryPropertyFiles(null));
+                scd = getServerConfigDocument(new CommonLogger(project), getLibertyDirectoryPropertyFiles(null));
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy:522]: Don't need to get props anymore at line 521.

**Review by cherylking — CHANGES_REQUESTED** [review]

File: src/main/groovy/io/openliberty/tools/gradle/tasks/UndeployTask.groovy (hunk 3 above)
Comment (arunvenmany-ibm) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/UndeployTask.groovy]: Added back logic in getServerDocument() to compare server.xml

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy (hunk 4 above)
Comment (arunvenmany-ibm) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DeployTask.groovy]: Added back logic in getServerDocument() to compare server.xml

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy (hunk 1 above)
Comment (arunvenmany-ibm) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy]: added this logic back. I think we should keep the logic

**Pull request closed without merging**